packages/storybook-ui-components/stories/Spinner.stories.tsx

```diff
@@ -1,36 +1,117 @@
 import React from "react";
-import { withKnobs, select } from "@storybook/addon-knobs";
-import { Spinner } from "@cockroachlabs/ui-components";
+import type { Meta, StoryObj } from "@storybook/react";
+import { Spinner, SpinnerSize, SpinnerFill } from "@cockroachlabs/ui-components";
 import { StoryContainer, StoryDescription  } from "./layout";
 
-export default {
+import { IconDisplaySection, IconDisplay, IconLabel, IconFrame } from "./Icon.stories";
+
+const sizes: Array<{ key: SpinnerSize; size: number }> = [
+  { key: "x-small", size: 16 },
+  { key: "small", size: 24 },
+  { key: "default", size: 40 },
+  { key: "large", size: 64 },
+];
+const fills: Array<SpinnerFill> = [
+  "danger",
+  "default",
+  "info",
+  "primary",
+  "success",
+  "warning",
+  "inverted",
+];
+
+const meta: Meta<typeof Spinner> = {
   title: "Spinner",
   component: Spinner,
-  decorators: [withKnobs],
+  tags: ["autodocs"],
+  decorators: [
+    (Story) => (
+      <StoryContainer>
+        <Story />
+      </StoryContainer>
+    ),
+  ],
+  argTypes: {
+    size: {
+      options: sizes.map(s => s.key),
+      control: "select",
+    },
+    fill: {
+      options: fills,
+      control: "select",
+    }
+  }
+};
+
+export default meta;
+
+type Story = StoryObj<typeof Spinner>;
+
+export const Demo: Story = {
+  args: {
+    size: "default",
+    fill: "default",
+  },
+  tags: ["autodocs"],
```

Comment:
IIRC you don't need to set `autodocs` on a particular story — it instead promotes the first story to be the "docs" story? I think? 🤔

Reply:
I was noticing a little weirdness in what was getting included in the "docs" story, so this might be why. Thanks for the hint.

Reply:
I think you are right about only including `tags: ["autodocs"]` in the _meta_ story. At some point I'll have to come back and think about the best way to organize these stories to get the cleanest view of the autodocs feature which seems nice. I'll do some reading up on what we get with v7.